src/Build/Graph/GraphBuilder.cs

```diff
@@ -307,41 +305,39 @@ private static void AddEdgesFromSolution(IReadOnlyDictionary<ConfigurationMetada
             // Project configurations are reused heavily, so cache the global properties for each
             Dictionary<string, ImmutableDictionary<string, string>> globalPropertiesForProjectConfiguration = new(StringComparer.OrdinalIgnoreCase);
 
+            IReadOnlyList<ProjectInSolution> projectsInSolution = solution.ProjectsInOrder;
             var newEntryPoints = new List<ProjectGraphEntryPoint>(projectsInSolution.Count);
 
             foreach (ProjectInSolution project in projectsInSolution)
             {
-                if (project.ProjectConfigurations.Count == 0)
+                if (!SolutionFile.IsBuildableProject(project))
                 {
                     continue;
                 }
 
                 ProjectConfigurationInSolution projectConfiguration = SelectProjectConfiguration(currentSolutionConfiguration, project.ProjectConfigurations);
 
-                if (projectConfiguration.IncludeInBuild)
+                if (!SolutionProjectGenerator.WouldProjectBuild(solution, currentSolutionConfiguration.FullName, project, projectConfiguration))
```

Comment:
I actually like using this method for slnf support. I don't think you currently support that anyway, but it'd be a nice future scenario to consider.

Reply:
Yep, I'll look into slnf in the future.